### DISCUSSION TITLE
Any modular way to create function objects for problem dependent program?

### DISCUSSION THREAD

**Discussion by sandeep026** [casadi/casadi#3485] [Q&A] [answered]
In my program I have a expression f which can depend on x, u, t and p. Depending on the problem this dependence and the number of input variables can vary.
Eg.
f(x,u,t,p)
f(x,u,t)
f(x,u,p)
f(x,u)

I want to create a function object which I will execute later but its creation now depends on the inputs. For now I handle this by using a if else statements to check dependency create the function for each possibilty. 

Is there simpler way to do this?

**Comment by jgillis** [comment, marked as answer]
I'm not sure what answer you are hoping for,
but you'd normally just list any possible dependencies, and add labels.
`F = Function('F',[f],[x,u,t,p],['F'],['x','u','t','p'])`

When the time comes to work with your function, you could use it with keyword arguments
`F(x=x,u=u)`
`F(x=x,u=u,p=p)`
...

**Reply by sandeep026** [reply]
I used a slight variant of the solution suggested. I coded a function which returns the list of variables in the expressions and its labels. Created the function object and called the function with the **dict.

"I'm not sure what answer you are hoping for,": I wanted to know if there is any feature like this in casadi. I found it convienient but not sure if caters for a larger audience. 

Pseudocode
x=SX.sym('x',1)
y=SX.sym('y',1)
f1=x+y+x**2
f2=x**2
create the map this way automatically by checking the dependencies
a=f1.function() # a: [x,y] --> [x+y+x**2]
b=f2.function() # b: [x] --> [x**2]
